arch/arc/core/swap.S

```diff
@@ -154,10 +154,6 @@ SECTION_FUNC(TEXT, __swap)
 .balign 4
 _swap_return_from_coop:
 
-	ld r1, [r2, _thread_offset_to_intlock_key]
-	st  0, [r2, _thread_offset_to_intlock_key]
-	ld r0, [r2, _thread_offset_to_return_value]
-
 	lr ilink, [_ARC_V2_STATUS32]
 	bbit1 ilink, _ARC_V2_STATUS32_AE_BIT, _return_from_exc
```

Comment:
I have some concern in branch _return_from_exc, as you move some codes.  these codes will not execute in _return_from_exc

Reply:
I know, that's why I asked about test coverage. I don't believe the return_from_exc path relies on r0/r1, does it?
What scenario would lead to the exception return path in swap_return_from_coop? What would it return to? I don;t see any context restore in that case at all.

Reply:
_return_from_exc path don't relay on r0/r1, but `st  0, [r2, _thread_offset_to_intlock_key]` not executed, not sure if it's matter.

Reply:
@IRISZZW  maybe we could just move this line "ld r0, [r2, _thread_offset_to_return_value]", because for now, the only problem is that the r0 value will be changed by sys_trace function, and the original r0 is relied on r2, so I think we could save r2 before sys_trace, and restore it to load original value into r0.